lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -285,6 +285,32 @@ defmodule ExUnit.Callbacks do
   end
 
   @doc """
+  Same as `start_supervised/2` but returns the PID on success and raises if
+  not started porperly.
+  """
+  @spec start_supervised!(Supervisor.child_spec | module | {module, term}, keyword) :: pid
+  def start_supervised!(child_spec_or_module, opts \\ []) do
+    case start_supervised(child_spec_or_module, opts) do
+      {:ok, pid} ->
+        pid
+      {:ok, pid, _info} ->
+        pid
+      {:error, reason} ->
+        raise """
+        Failed to start child with the spec #{inspect child_spec_or_module}.
+        Reason: #{start_supervised_error(reason)}
+        """
+    end
+  end
+
+  defp start_supervised_error({{:EXIT, reason}, _info}),
+    do: Exception.format_exit(reason)
+  defp start_supervised_error({reason, _info}),
+    do: Exception.format_exit(reason)
+  defp start_supervised_error(reason),
+    do: Exception.format_exit(reason)
```

Comment:
We have three different clauses here but two tests. :) Which one is missing? Can we test the missing one?

Reply:
Nice catch, I'll se if the invalid child spec falls on this one.

Reply:
Well, a child spec gave me an "ArgumentError" exception, no error tuple is returned from `start_supervised` in this case, and that was the previous behaviour by the way.

I can't think in any other case that matches this last clause, and [the spec](https://hexdocs.pm/elixir/Supervisor.html#t:on_start_child/0) of `Supervisor.start_child/2` declares that the return of this function can be:

```elixir
@type on_start_child ::
  {:ok, child} |
  {:ok, child, info :: term} |
  {:error, {:already_started, child} | :already_present | term}
``` 

The reason I've made this last clause is to handle this possible unknown error reasons. What do you think? Leave or remove it?

Reply:
PS.: the missing one is the last one.